distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1424,54 +1447,244 @@ public void testProductDownloadListEmptyWhenNoDownloads() {
+
+    }
+
+    /**
+     * Helper method used to determine that certain messages are showing up in the cometDClient notifications.
+     * @param cometDClient the CometDClient to check for messages.
+     * @param filename1 the filename of the first resource to check against the CometDClient.
+     * @param filename2 the filename of the second resource to check against the CometDClient.
+     * @param messageToFind1 the message to find in the CometDClient notifications related to the first filename.
+     * @param messageToFind2 the message to find in the CometDClient notifications related to the second filename.
+     * @return a boolean that is only true when both messages have been found in the notifications.
+     */
+    private boolean foundExpectedNotifications(CometDClient cometDClient, String filename1, String filename2, String messageToFind1, String messageToFind2)
+    {
+        boolean foundCorrectMessages = false;
+        List<String> notifications = cometDClient.getMessages(NOTIFICATIONS_CHANNEL);
+
+
+        int numberOfThingsToFind = 2;
+
+        int numberOfThingsFound = 0;
+
+        for(int index = 0; index < notifications.size(); index++) {
+
+            LOGGER.debug("This is what is at the notifications index: {}", notifications.get(index).toString());
+
+
+           if(notifications.get(index).toString().contains(messageToFind1) && notifications.get(index).toString().contains(filename1) && !messageToFind1.equals(""))
+            {
+                numberOfThingsFound++;
+            }
+            else if(notifications.get(index).toString().contains(messageToFind2) && notifications.get(index).toString().contains(filename2) && !messageToFind2.equals(""))
+            {
+                numberOfThingsFound++;
+            }
+
+
+        }
+
+
+        if (numberOfThingsFound == numberOfThingsToFind)
+        {
+            foundCorrectMessages = true;
+        }
+
+        return foundCorrectMessages;
+    }
+
+    /**
+     * Helper method used to determine that certain messages are showing up in the cometDClient activities.
+     * @param cometDClient the CometDClient to check for messages.
+     * @param filename1 the filename of the first resource to check against the CometDClient.
+     * @param filename2 the filename of the second resource to check against the CometDClient.
+     * @param messageToFind1 the message to find in the CometDClient activities related to the first filename.
+     * @param messageToFind2 the message to find in the CometDClient activities related to the second filename.
+     * @return a boolean that is only true when both messages have been found in the activities.
+     */
+    private boolean foundExpectedActivities(CometDClient cometDClient, String filename1, String filename2, String messageToFind1, String messageToFind2)
+    {
+       boolean foundCorrectActivities = false;
+        List<String> activities = cometDClient.getMessages(ACTIVITIES_CHANNEL);
```

Comment:
f you are expecting messages in ascending order, you probably want to use cometDClient.getMessagesInAscOrder(ACTIVITIES_CHANNEL)